c2rust-analyze/src/context.rs

```diff
@@ -1399,3 +1399,127 @@ impl Assignment<'_> {
         )
     }
 }
+
+/// Print an `LTy` as a string, using the provided callback to print the labels on each pointer and
+/// reference type.
+///
+/// Note this completely omits any labels on non-pointer types.
+pub fn print_ty_with_pointer_labels<L: Copy>(
+    lty: LabeledTy<L>,
+    mut f: impl FnMut(L) -> String,
+) -> String {
+    let mut out = String::new();
+    print_ty_with_pointer_labels_into(&mut out, lty, &mut f);
+    out
+}
+
+pub fn print_ty_with_pointer_labels_into<L: Copy>(
+    dest: &mut String,
+    lty: LabeledTy<L>,
+    f: &mut impl FnMut(L) -> String,
```

Comment:
Does this need to be passed by reference here, when it's by value in the other function? Is it because of the recursion?

Reply:
Yes, this is the easiest way to avoid issues with the branching recursion.